sidecar/proxy/envoy/manager.go

```diff
@@ -292,24 +292,62 @@ func ParseServiceKey(s string) (string, []string) {
 }
 
 func buildClusters(rules []api.Rule) []Cluster {
-	clusterMap := make(map[string]struct{})
+	clusterMap := make(map[string]*api.Backend)
 	for _, rule := range rules {
 		if rule.Route != nil {
 			for _, backend := range rule.Route.Backends {
 				key := buildServiceKey(backend.Name, backend.Tags)
-				clusterMap[key] = struct{}{}
+				clusterMap[key] = &backend
 			}
 		}
 	}
 
 	clusters := make([]Cluster, 0, len(clusterMap))
-	for clusterName := range clusterMap {
+	for clusterName, backend := range clusterMap {
+
 		cluster := Cluster{
 			Name:             clusterName,
 			ServiceName:      clusterName,
 			Type:             "sds",
 			LbType:           "round_robin",
 			ConnectTimeoutMs: 1000,
+			Hystrix:          CircuitBreaker{},
+			OutlierDetection: OutlierDetection{
+				MaxEjectionPercent: 100,
+			},
+		}
+
+		if backend.Hystrix != nil {
+			if backend.Hystrix.TimeoutMS > 0 {
+				// convert from float sec to int ms
+				cluster.ConnectTimeoutMs = int(backend.Hystrix.TimeoutMS * 1000)
```

Comment:
On second thought, I take it back. Lets revert to using consistent "timeout".. and eliminate all the _ms and MS stuff. In an future PR, we can revisit changing all these to milliseconds.